Modules/Filtering/FFT/include/itkFFTImageFilterFactory.h

```diff
@@ -116,6 +117,9 @@ class FFTImageFilterFactory : public itk::ObjectFactoryBase
     OverrideFFTImageFilterType<typename FFTImageFilterTraits<TFFTImageFilter>::template InputPixelType<float>,
                                typename FFTImageFilterTraits<TFFTImageFilter>::template OutputPixelType<float>,
                                3>();
+    OverrideFFTImageFilterType<typename FFTImageFilterTraits<TFFTImageFilter>::template InputPixelType<float>,
```

Comment:
`ITKMontage` tests require FFT padding for a 4D image, while FFT factory overrides were previously only registered for 1D, 2D, and 3D images. After consideration I think it's reasonable to add a default override for Vnl and FFTW FFT implementations for 4D images as this is a reasonably commonly used ITK image dimension and appears to be supported by Vnl and FFTW backends. 

For higher dimensions we will require consuming projects to explicitly register FFT factory overrides for their needs. Other FFT implementations in remote modules such as ITKVkFFTBackend may not support 4D images, so I expect to revisit this in the near future to make 4D overrides optional.

Reply:
Sounds good.